### PR TITLE
Add flight routines python module with examples

### PR DESCRIPTION
**Pull request by goldarte**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+.hypothesis/
+.pytest_cache/
+
+# Translations
+*.mo
+*.pot
+
+# Django stuff:
+*.log
+local_settings.py
```
Comment (okalachev) [inline .gitignore]: What for all of this here?

File: README.md [hunk 2]
```diff
@@ -1,9 +1,190 @@
...
+```bash
+cd ~/catkin_ws/src
+git clone https://github.com/CopterExpress/clever_tools.git
+```
+
+Execute catkin_make in catkin workspace directory:
+
+```bash
+cd ~/catkin_ws
+catkin_make clever_tools
```
Comment (okalachev) [inline README.md]: Are you sure this is gonna work? After `catkin_make` there should be the make target, do we have `clever_tools` target?

File: README.md [hunk 3]
```diff
@@ -1,9 +1,190 @@
...
+
+## Python modules
+
+### flight_routines
+
+#### Usage example
+
+```
+import rospy
+from flight_routines import *
```
Comment (okalachev) [inline README.md]: Actually, importing star considered bad practice. When analysing the code, it's very difficult to tell, where from come all the identifiers.

File: README.md [hunk 4]
```diff
@@ -1,9 +1,190 @@
...
+* `set_rates` to service `/set_rates`
+* `get_telemetry` to service `get_telemetry`
+* `land` to service `/land`
+* `arming` to service `/mavros/cmd/arming`
+* `set_mode` to service `/mavros/set_mode`
+
+#### Functions
+
+```bash
+get_distance(x1, y1, z1, x2, y2, z2)
```
Comment (okalachev) [inline README.md]: Shouldn't it be a list of something? Otherway this is just code samples within the text.

File: clever_tools/setup.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from distutils.core import setup
+from catkin_pkg.python_setup import generate_distutils_setup
+ 
+setup_args = generate_distutils_setup(
+    packages=['flight_routines'],
```
Comment (okalachev) [inline clever_tools/setup.py]: Shouldn't `flight_routines` be submodules from `clever_tools`? Why they should be seen as a global package?

File: clever_tools/src/flight_routines/__init__.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    result = navigate(x=start.x, y=start.y, z=start.z+height, speed=speed, yaw=float('nan'), frame_id=frame_id, auto_arm=True)
+    rospy.loginfo(result)
+    while abs(climb - height) > tolerance:
+        climb = abs(get_telemetry(frame_id=frame_id).z - start.z)
+        rospy.loginfo("Takeoff to {:.2f} of {:.2f} meters".format(climb, height))
+        rate.sleep()
+    rospy.loginfo("Takeoff succeeded!")
+
+# Reach specified point. Copter needs to be armed.
+def reach_point(x, y, z, yaw=float('nan'), speed=SPEED, tolerance=TOLERANCE, frame_id=LOCAL_FRAME_ID):
```
Comment (okalachev) [inline clever_tools/src/flight_routines/__init__.py]: Actually, we have a more universal version of this function, utilizing `navigate_target` frame: see https://clever.copterexpress.com/ru/snippets.html#block-nav.

File: .gitignore (hunk 1 above)
Comment (goldarte) [reply .gitignore]: It's a default .gitignore file for python projects: https://github.com/github/gitignore/blob/master/Python.gitignore

Comment (okalachev) [reply .gitignore]: But who made it the default? `.gitignore` is a project specific-file, not language-specific.

File: README.md (hunk 2 above)
Comment (goldarte) [reply README.md]: Yes, because we have CMakeList for it.

File: README.md (hunk 3 above)
Comment (goldarte) [reply README.md]: Changed this in https://github.com/CopterExpress/clever_tools/pull/2/commits/9a171f9f7992e75b708a2f44de95b9eaddfa2d2d

File: .gitignore (hunk 1 above)
Comment (goldarte) [reply .gitignore]: Simplified it in https://github.com/CopterExpress/clever_tools/pull/2/commits/636cffa81c69638f42c55e536cf571213faa1afe

File: README.md (hunk 4 above)
Comment (goldarte) [reply README.md]: I don't understand this

File: clever_tools/CMakeLists.txt [hunk 7]
```diff
@@ -33,7 +33,7 @@ find_package(catkin REQUIRED COMPONENTS
 ## Uncomment this if the package has a setup.py. This macro ensures
 ## modules and global scripts declared therein get installed
 ## See http://ros.org/doc/api/catkin/html/user_guide/setup_dot_py.html
-# catkin_python_setup()
+## catkin_python_setup()
```
Comment (okalachev) [inline clever_tools/CMakeLists.txt]: What these changes for?

File: clever_tools/CMakeLists.txt [hunk 8]
```diff
@@ -170,10 +170,10 @@ target_link_libraries(publish_frame ${catkin_LIBRARIES})
 
 ## Mark executable scripts (Python etc.) for installation
 ## in contrast to setup.py, you can choose the destination
-# install(PROGRAMS
-#   scripts/my_python_script
-#   DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-# )
+## install(PROGRAMS
```
Comment (okalachev) [inline clever_tools/CMakeLists.txt]: Still irrelevant changes.

**Review by okalachev — CHANGES_REQUESTED** [review]

**Pull request merged**